platform/darwin/src/MGLStyleValue.mm

```diff
@@ -62,16 +62,59 @@ - (NSUInteger)hash {
 
 @end
 
+@implementation MGLStyleFunction
+
++ (instancetype)functionWithStops:(NSDictionary *)stops {
+    return [[self alloc] initWithInterpolationBase:1.0 stops:stops];
+}
+
++ (instancetype)functionWithInterpolationBase:(CGFloat)interpolationBase stops:(NSDictionary *)stops {
+    return [[self alloc] initWithInterpolationBase:interpolationBase stops:stops];
+}
+
+- (instancetype)init {
+    return [self initWithInterpolationBase:1.0 stops:@{}];
+}
+
+- (instancetype)initWithInterpolationBase:(CGFloat)interpolationBase stops:(NSDictionary *)stops {
+    if (self = [super init]) {
+        self.interpolationBase = interpolationBase;
+        self.stops = stops;
+    }
+    return self;
+}
+
+- (NSString *)description {
+    return [NSString stringWithFormat:@"<%@: %p, \
+            stops = %@, \
+            interpolationBase = %f>",
+            NSStringFromClass([self class]), (void *)self,
+            self.stops,
+            self.interpolationBase];
+}
+
+- (BOOL)isEqual:(MGLStyleFunction *)other {
+    return ([other isKindOfClass:[self class]]
+            && [other.stops isEqualToDictionary:self.stops]
+            && other.interpolationBase == self.interpolationBase);
+}
+
+- (NSUInteger)hash {
+    return  self.stops.hash + self.interpolationBase;
```

Comment:
Use `@(self.interpolationBase).hash`, not `interpolationBase` itself, to further minimize collisions. There are similar problems with `interpolationMode` in the subclasses. Feel free to take care of this in a separate PR.